src/main/webapp/app/text-assessment/text-assessment.component.ts

```diff
@@ -251,6 +255,31 @@ export class TextAssessmentComponent implements OnInit, OnDestroy, AfterViewInit
             });
         }
     }
+    /**
+     * Load next assessment in the same page.
+     */
+    assessNextOptimal() {
+        if (this.exercise.type === ExerciseType.TEXT) {
+            this.textSubmissionService.getTextSubmissionForExerciseWithoutAssessment(this.exercise.id).subscribe(
+                (response: TextSubmission) => {
+                    this.unassessedSubmission = response;
+                    this.router.onSameUrlNavigation = 'reload';
+                    // navigate to root and then to new assessment page to trigger re-initialization of the components
+                    this.router
+                        .navigateByUrl('/', { skipLocationChange: true })
```

Comment:
Works, but a workaround :)
As far as I understand it, this will trigger the else clause of the ngOnInit (127 - 130).
So this section could be refactored into a method which is then also called by your new method.

**Why we should change it in my opinion:**
- The solution needs a comment to be understood as it is a workaround for a proper solution
- There are unnecessary interactions with the accountService if ngOnInit is called again
- Somebody could incidentally break your implementation when changing the ngOnInit method and has no idea that you are implicitly calling it
- We have another place where the route as a string is constructed (could be overlooked if we want to change the url)

Reply:
@fde312 
I have a better idea on how to solve this issue, so it would be even clearer.

You could use a param subscription:
`this.paramSub = this.route.params.subscribe(params => {
        const exerciseId = Number(params.get('exerciseId'));
}`

This way you could update both the url and have an obvious reload mechanism.
(Otherwise the url would not change which is a valid point in your implementation)